src/App.tsx

```diff
@@ -1,26 +1,195 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { TodoList } from './components/TodoList/TodoList';
+import { deleteTodos, getTodos, updateTodos, USER_ID } from './api/todos';
+import { Footer } from './components/Footer/Footer';
+import { Status } from './types/Status';
+// eslint-disable-next-line max-len
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [statusOfTodos, setStatusOfTodos] = useState(Status.all);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingIds, setloadingIds] = useState<number[]>([]);
+  const [deletingTodoId, setDeletingTodoId] = useState<number | null>(null);
+  //const [isEditing, setIsEdiding] = useState(false);
+
+  function loadTodos() {
+    setIsLoading(true);
+
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'))
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }
+
+  if (errorMessage) {
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  }
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  let filteredTodos = todos;
+
+  if (statusOfTodos === Status.active) {
+    filteredTodos = filteredTodos.filter(todo => !todo.completed);
+  }
+
+  if (statusOfTodos === Status.completed) {
+    filteredTodos = filteredTodos.filter(todo => todo.completed);
+  }
```

Comment:
Suggest to move logiс for filtering todo in funciton. It can be as util outside of component,  will be accept todos and statusOfTodo and return filteredTodos.